src/realm/sync/subscriptions.cpp

```diff
@@ -492,6 +470,19 @@ util::Future<SubscriptionSet::State> SubscriptionSet::get_state_change_notificat
     return std::move(future);
 }
 
+void SubscriptionSet::get_state_change_notification(
+    State notify_when, util::UniqueFunction<void(util::Optional<State>, util::Optional<Status>)> cb) const
```

Comment:
This matches our normal API for async callbacks of `void(optional_or_nullable<ReturnType>, optional_or_nullable<ErrorType>)`. If you want, I can put this on `Future` instead.

Reply:
If it makes you feel better no human will ever actually use this signature. The binding generator transforms it to `function getStateChangeNotification(notify_when: SubscriptionSetState) -> Promise<SubscriptionSetState>` (or whatever the native async signature would be in your language of choice) which is consumed via `await set.getStateChangeNotification(blah)`. This signature is only used as part of the mechanical binding process.

If you are curious, the signature transform logic is [here](https://github.com/realm/realm-js/blob/e04e56b679786679641abb195d5b7cc3ae44b73e/packages/bindgen/src/bound-model.ts#L169-L210) (this is language agnostic), and the asyncification happens [here](https://github.com/realm/realm-js/blob/e04e56b679786679641abb195d5b7cc3ae44b73e/packages/bindgen/src/templates/node-wrapper.ts#L41-L59) via the call from [here](https://github.com/realm/realm-js/blob/e04e56b679786679641abb195d5b7cc3ae44b73e/packages/bindgen/src/templates/node-wrapper.ts#L113) (this is js-specific but shared by both engines). Interestingly, the actual generated C++ code doesn't care about the transform and just takes a normal callback from JS. This is part of why I didn't want to do a one-off for Future support just for this class because that strategy would need to change. I don't know if that is the implementation strategy all languages will choose, but it seems to work well here at least.